tensorflow_probability/python/mcmc/transformed_kernel.py

```diff
@@ -110,6 +106,33 @@ def transformed_log_prob_fn(*state_parts):
   return transformed_log_prob_fn
 
 
+def target_log_prob_getter(kernel):
+  kernel_stack = [kernel]
+  while 'target_log_prob_fn' not in kernel.parameters:
+    if 'inner_kernel' not in kernel.parameters:
+      raise ValueError('"None of the nested `inner_kernel`s contains a '
+                       '`target_log_prob_fn`."')
+    kernel = kernel.inner_kernel
+    kernel_stack.append(kernel)
+  return kernel.parameters['target_log_prob_fn'], kernel_stack
+
+
+def target_log_prob_setter(kernel_stack, new_target_log_prob):
+  # edit the appropriate kernel with the new log_prob_fn
+  last_kernel = kernel_stack.pop()
+  last_kernel_kwargs = last_kernel.parameters.copy()
+  last_kernel_kwargs.update(target_log_prob_fn=new_target_log_prob)
+
+  # propogate upwards
+  while kernel_stack:
+    curr_kernel = kernel_stack.pop()
+    with deprecation.silence():
+      curr_kernel._inner_kernel = type(last_kernel)(**last_kernel_kwargs)
+    last_kernel = curr_kernel
+    last_kernel_kwargs = last_kernel.parameters
+
+  return last_kernel, last_kernel_kwargs
```

Comment:
This function will also be much simpler to use and understand if it relies on returning the final reconstructed kernel rather than on mutating the `_inner_kernel` field of the top-level kernel, as it currently appears to do.  So, I suggest changing the API of this helper so the call site in `TransformedTransitionKernel` looks like
```
self._inner_kernel = update_target_log_prob(self._inner_kernel, new_target_log_prob)
```

Reply:
Fixed.